es6-shim.js

```diff
@@ -291,6 +291,13 @@
       return x != null && Object(x) === x;
     },
 
+    TypeIsNotObject: function (val) {
+      // This is faster when TypeIsObject may commonly return false.
+      return val === void 0 || val === null || val === true || val === false ||
```

Comment:
> my main concern is that new primitive typeof values (which there will likely be many) will require a whitelist approach to be continually updated, whereas a blacklist approach will work essentially forever.

Actually the opposite -- see the last case in http://www.ecma-international.org/ecma-262/6.0/#sec-typeof-operator-runtime-semantics-evaluation .  New exotic objects will not require this explicit list to be redefined.

Of course, if a new *primitive* type is added (like `Symbol`), all bets are off.  But not much we can do about that, other than update es6-shim when that day comes.